### PR TITLE
Add GlobalPreferencesController and GlobalPreferencesFetcher

### PR DESCRIPTION
**Pull request by joewalsh**
data model for https://phabricator.wikimedia.org/T220549

### CODE REVIEW

File: WMF Framework/Fetcher.swift [hunk 1]
```diff
@@ -200,7 +199,7 @@ public class Token: NSObject {
     }
 }
 
-public enum FetcherResult<Success, Error> {
+public enum Result<Success, Error> {
```
Comment (joewalsh) [inline WMF Framework/Fetcher.swift]: Should we move this to a separate file and use it everywhere we want a result type?

Comment (montehurd) [reply WMF Framework/Fetcher.swift]: That sounds cool! 😎

Comment (tonisevener) [reply WMF Framework/Fetcher.swift]: Can we just use the Swift 5 Result type?

Comment (joewalsh) [reply WMF Framework/Fetcher.swift]: good point - forgot that the Result type landed with Swift 5. Will remove this one.

**Comment by joewalsh** [comment]
setting WIP to work out global/local preference questions

File: Wikipedia/Code/WMFAppViewController.m [hunk 2]
```diff
@@ -252,18 +252,21 @@ - (NSURL *)siteURL {
...
     [self.periodicWorkerController add:[WMFEventLoggingService sharedInstance]];
+    [self.periodicWorkerController add:self.dataStore.globalPreferencesController];
 
     self.backgroundFetcherController = [[WMFBackgroundFetcherController alloc] init];
     self.backgroundFetcherController.delegate = self;
     [self.backgroundFetcherController add:self.dataStore.readingListsController];
     [self.backgroundFetcherController add:self.dataStore.remoteNotificationsController];
     [self.backgroundFetcherController add:(id<WMFBackgroundFetcher>)self.dataStore.feedContentController];
     [self.backgroundFetcherController add:[WMFEventLoggingService sharedInstance]];
+    [self.periodicWorkerController add:self.dataStore.globalPreferencesController];
```
Comment (tonisevener) [inline Wikipedia/Code/WMFAppViewController.m]: should be backgroundFetcherController :)

Comment (joewalsh) [reply Wikipedia/Code/WMFAppViewController.m]: thanks! fixed

**Pull request merged**